.github/workflows/pr-checks.yaml

```diff
@@ -2,103 +2,49 @@ name: Pull request checks
 
 on: [pull_request]
 
-jobs:
-  build-environment:
-    runs-on: ubuntu-18.04
-    name: Build conda environment
-    steps:
-      - uses: actions/checkout@v2
-        name: Checkout repository
-
-      # See: https://github.com/marketplace/actions/setup-conda
-      - name: Setup anaconda
-        uses: s-weigand/setup-conda@v1
-        with:
-          conda-channels: "conda-forge"
-
-      # Build cache of environment
-      - name: Cache conda environment
-        id: cache-environment
-        uses: actions/cache@v2
-        with:
-          path: nxviz.tar.gz
-          # Note: Remember that whatever files the environment build depends on
-          # should be hashed and added to the key.
-          key: ${{ runner.os }}-env.${{ hashFiles('environment.yml') }}
-
-      - name: Build environment
-        if: steps.cache-environment.outputs.cache-hit != 'true'
-        run: |
-          conda env create -f environment.yml
-          python -m pip install .
-
-      - name: Install conda-pack
-        if: steps.cache-environment.outputs.cache-hit != 'true'
-        run: conda install -c conda-forge conda-pack
-
-      - name: Run conda-pack
-        if: steps.cache-environment.outputs.cache-hit != 'true'
-        run: conda pack -n nxviz -o nxviz.tar.gz
-
-      # See: https://github.com/actions/upload-artifact
-      - name: Upload environment
-        uses: actions/upload-artifact@v2
-        with:
-          name: nxviz-tarball
-          path: nxviz.tar.gz
+env:
+  cache-version: "cache-v2"
 
-  unit-tests:
-    name: Run unit tests
+jobs:
+  pre-commit:
     runs-on: ubuntu-latest
-    needs: build-environment
-
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+      - uses: pre-commit/action@v3.0.1
+  build-environment:
+    needs: pre-commit
+    strategy:
+      max-parallel: 10
+      matrix:
+        environment: [py310, py311, py312, py313]
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    name: Build and test
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v2
-
-      # https://github.com/actions/download-artifact
-      - name: Download environment tarball
-        uses: actions/download-artifact@v2
+        uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
```

Comment:
I think `setup-node` might not be necessary, based on what I'm seeing in the rest of the pipeline.

Reply:
i thought so too but then i got a weird crash with a macos node not having `make`.

Reply:
Ah, I see! That makes sense now, I guess `node` also installs `make`. Could you leave a comment just above the line that explains this?